framework/include/linesearches/LineSearch.h

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef LINESEARCH_H
+#define LINESEARCH_H
+
+#include "libmesh/parallel_object.h"
+#include "ConsoleStreamInterface.h"
+
+using namespace libMesh;
+
+class FEProblemBase;
+
+class LineSearch : public ConsoleStreamInterface, public ParallelObject
+{
+public:
+  LineSearch(FEProblemBase & fe_problem, MooseApp & app);
+
+  // A dummy pure virtual destructor because this must be a base class; the derived class must
+  // implement the line searching capabilities
+  virtual ~LineSearch() = 0;
```

Comment:
No, this isn't the way to do this. Yes you do need to declare a virtual destructor, but it should not be pure virtual. In fact we _don't_ want people to implement destructors unless they have something to destruct.